server.js

```diff
@@ -66,55 +65,41 @@ app.use('/', (req, res, next) => {
   return next();
 });
 
+app.use('/*', (req, res, next) => {
+  const initialStore = { ...initialState, lastLoaded: req._parsedUrl.path };
+  global.store = configureStore(initialStore);
```

Comment:
This might be a bit confusing because `global` isn't a global variable in Nodejs. I think it's just global by default since it's not declared. It might be better to explicitly declare this variable (outside this function) so you can reuse it in this `.use` function and later in `.get`.

It brings up the question, however, is the `store` unique per user? If the `store` is being stored globally in the Nodejs initial server file, it's possible that multiple sessions can have the same store data. Can you verify that this _isn't_ happening?

Reply:
I think I originally wrote this so I'll respond:

1. Thanks for pointing that out about `global`, I didn't realize it was just a convention. We could use `store` instead. We could also try to make it a declared variable but the advantage of making it global is that it can be used easily in the methods called in `ApiRoutes`.
2. The `store` should be unique per user. The mechanism to enforce this is that we reset `global.store` to a fresh instance (`configureStore(initialStore)`) in each call.

Reply:
Do either of you know where/how global is originally set? Otherwise, `global.store = configureStore(initialStore);` would throw an error I thought.

Reply:
Edwin can correct if this is wrong, but I thought `global` was the Node name for the global object, e.g. `window` in the browser. If you start a node session and enter `global`, you get `Object [global] { ... }`

Reply:
Ah, yes, `global` is the actual name for the global object in Node.

Reply:
Yeah, that makes sense for 1. and 2. Personally, I don't like using global variables if I can help it but I see how it can help in this case for other functions in `ApiRoutes`.